SecureAuth.Sdk/NumberProfileService.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Linq;
+
+namespace SecureAuth.Sdk
+{
+    public class NumberProfileService : INumberProfileService
+    {
+        private readonly ApiClient _apiClient;
+
+        protected internal NumberProfileService(ApiClient apiClient)
+        {
+            this._apiClient = apiClient;
+        }
+
+        /// <summary>
+        /// Evaluate the provided phone number for carrier status and details.
+        /// </summary>
+        /// <param name="userId">The unique user ID.</param>
+        /// <param name="phoneNumber">The phone number for evaluation.</param>
+        /// <returns>NumberProfileResponse</returns>
+        public NumberProfileResponse EvaluatePhoneNumber(string userId, string phoneNumber)
```

Comment:
I think this is my golang leaking, just a design decision to add a helper method that makes it a bit easier for the implementer, if they didn't want to create the NumberProfileRequest object. The NumberProfileRequest is used both in the update and get functions, so it could be confusing.